src/gui/sharee.cpp

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (C) by Roeland Jago Douma <roeland@owncloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#include "sharee.h"
+#include "ocsshareejob.h"
+
+namespace OCC {
+
+Sharee::Sharee(const QString shareWith,
+               const QString displayName,
+               const Type type)
+: _shareWith(shareWith),
+  _displayName(displayName),
+  _type(type)
+{
+}
+    
+QString Sharee::shareWith() const
+{
+    return _shareWith;
+}
+
+QString Sharee::displayName() const
+{
+    return _displayName;
+}
+
+Sharee::Type Sharee::type() const
+{
+    return _type;
+}
+
+
+ShareeModel::ShareeModel(AccountPtr account,
+                         const QString search,
+                         const QString type,
+                         QObject *parent)
+: QAbstractTableModel(parent),
+  _account(account),
+  _search(search),
+  _type(type)
+{
+    OcsShareeJob *job = new OcsShareeJob(_account, this);
+    connect(job, SIGNAL(shareeJobFinished(QVariantMap)), SLOT(shareesFetched(QVariantMap)));
+    job->getSharees(_search, _type, 1, 50);
+}
+
+void ShareeModel::shareesFetched(const QVariantMap &reply)
+{
+    qDebug() << Q_FUNC_INFO << reply;
+
+    auto data = reply.value("ocs").toMap().value("data").toMap();
+
+    QVector<QSharedPointer<Sharee>> newSharees;
```

Comment:
FYI, I think some compilers don't like ```>>``` and you need to write it as ```QVector<QSharedPointer<Sharee> >``` ?
@ogoffart